src/diagnostics/MOM_spatial_means.F90

```diff
@@ -167,27 +165,40 @@ function global_area_integral(var, G, scale, area, tmp_scale, unscale)
-                                                         !! variable that is reversed in the return value [a A-1 ~> 1]
+  real,                   optional, intent(in)  :: tmp_scale !< A temporary rescaling factor for the variable
+                                                         !! that is reversed in the return value [a A-1 ~> 1],
+                                                         !! or [b B-1 ~> 1] if unscale is also present.
   real,                   optional, intent(in)  :: unscale !< A rescaling factor for the variable [a A-1 ~> 1]
                                                          !! that converts it back to unscaled (e.g., mks)
-                                                         !! units to enable the use of the reproducing sums.
+                                                         !! units to enable the use of the reproducing sums, or
+                                                         !! a factor converting between rescaled units if
+                                                         !! tmp_scale is also present [B A-1 ~> b a-1].
                                                          !! Here scale and unscale are synonymous, but unscale
                                                          !! is preferred and takes precedence if both are present.
   real :: global_area_integral !< The returned area integral, usually in the units of var times an area,
-                               !! [a m2] or [A m2 ~> a m2] depending on which optional arguments are provided
+                               !! [a m2] or [A L2 ~> a m2] or [B L2 ~> b m2], depending on which optional
+                               !! arguments are provided
 
   ! Local variables
-  ! In the following comments, [A] is used to indicate the arbitrary, possibly rescaled units of the
-  ! input array while [a] indicates the unscaled (e.g., mks) units that can be used with the reproducing sums
-  real, dimension(SZI_(G),SZJ_(G)) :: tmpForSumming ! An unscaled cell integral [a m2]
-  real :: scalefac  ! An overall scaling factor for the areas and variable, perhaps in [m2 a A-1 L-2 ~> 1]
-  real :: temp_scale ! A temporary scaling factor [a A-1 ~> 1] or [1]
+  ! In the following comments, [A ~> a] is used to indicate the arbitrary, possibly rescaled units of the
+  ! input array while [a] indicates the unscaled (e.g., mks) units that can be used with the reproducing sums.
+  ! [A ~> a] and [B ~> b] are the same units unless tmp_scale and unscale are both present.
+  real, dimension(SZI_(G),SZJ_(G)) :: tmpForSumming ! An unscaled cell integral in [a m2] or [B L2 ~> b m2]
+                    ! or [m2] or [L2 ~> m2] if var is not present.
+  real :: scalefac  ! An overall scaling factor for the areas and variable, in units of [a m2 A-1 L-2 ~> 1]
+                    ! or [1] or [B m2 A-1 L-2 ~> b a-1] or [B A-1 ~> b a-1] depending on which
+                    ! optional arguments are present.
```

Comment:
I think there are still a few errors in the units. Note that var is not an optional argument. Below is what I think is correct for crosschecking.

|`tmp_scale` | `unscale` | `scalefac` | `tmpForSumming` |
|-|-|-|-|
|T |T |[B A-1 ~> b a-1] |[B L2 ~>  b m2] |
|T |F |[1] |[A L2 ~> a m2] |
|F |T |[a m2 A-1 L-2] |[a m2] |
|F |F |[m2 L-2] |[A m2 ~> a m2] |

Reply:
A version of this table has been added in comments.